Lib/test/test_concurrent_futures.py

```diff
@@ -342,6 +342,31 @@ def test_hang_issue12364(self):
         for f in fs:
             f.result()
 
+    def test_cancel_futures(self):
```

Comment:
Could you test some more scenarios? Like when `wait` is also true? When called and the interpreter exits.

Reply:
I addressed the part about `wait` in https://github.com/python/cpython/pull/18057#issuecomment-577011294, but I'm not certain about how to write tests that account for the interpreter exiting. Could you elaborate on what that might look like? I'd certainly be glad to add some additional test coverage.

Reply:
I have some code on another computer that I can send as an example. I'll try to add it to this PR by Monday.

Reply:
I was imagining something like:

```python
    def testXXX(self):
        rc, out, err = assert_python_ok('-c', """if True:
            from concurrent.futures import {executor_type}
            from test.test_concurrent_futures import sleep_and_print
            if __name__ == "__main__":
                t = {executor_type}(max_workers=3)
                t.submit(sleep_and_print, 1.0, "apple")
                t.shutdown(wait=False, cancel_futures=True)
            """.format(executor_type=self.executor_type.__name__))
        # Errors in atexit hooks don't change the process exit code, check
        # stderr manually.
        self.assertFalse(err)
        self.assertEqual(out.strip(), b"apple")
```

Note that this might hang with ProcessPoolExecutor (it does withoutout `cancel_futures`).

Reply:
> Note that this might hang with ProcessPoolExecutor (it does withoutout cancel_futures).

Yeah from what I can tell, ProcessPoolExecutor doesn't function reliably with `wait=False` (regardless of `cancel_futures`) when there are any futures currently running in the executor, but it works fine for ThreadPoolExecutor. So, I'll just add something like this to the ThreadPoolExcutor shutdown tests. Thanks for the example.